### ISSUE TITLE
[P0] STOP THE LINE: Payload tracing is STUBBED - cannot debug garbage output (APR-TRACE-001)

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#154]
## Summary

**STOP THE LINE (Jidoka)**: The `apr trace --payload` command is a NON-FUNCTIONAL STUB. When GGUF inference produces garbage output, we have NO WAY to see what went in, which layer processed it, and what came out.

This is a P0 fatal defect blocking all debugging of the GGUF garbage output issue (LAYOUT-001 regression).

### Stub Location: `crates/apr-cli/src/commands/trace.rs:197-199`

```rust
if payload {
    println!("Tracing payload through model: {}", path.display());
}
```

**This just prints a message and continues to the non-payload flow. It does NOTHING.**

### InferenceTracer Exists But Is NOT Integrated

The `realizar/src/inference_trace/mod.rs` has a comprehensive `InferenceTracer` with:
- `trace_encode()` - tokenization step
- `trace_embed()` - embedding lookup
- `trace_layer()` - transformer layer
- `trace_lm_head()` - logits projection
- `trace_sample()` - token sampling
- `trace_decode()` - token decoding

**BUT** the GGUF generation path (`gguf_monolith.rs:10777 generate_with_cache`) has ZERO tracer integration:

```bash
$ grep -r "tracer\.\|InferenceTracer\|trace_layer" realizar/src/gguf/
# NO MATCHES
```

### Current Debugging Experience

When GGUF outputs garbage like "random random Domain Domain pandas random":
1. We cannot see the input tokens that went in
2. We cannot see which layer produced bad activations
3. We cannot see the logits before sampling
4. We cannot see the token IDs being decoded
5. **WE ARE FLYING BLIND**

## Spec Reference

APR-SPEC.md lines 655-685 specifies payload tracing:

```
apr trace model.apr --input test.wav --trace-payload

Step 1: Audio Input
  Shape: [1, 480000]  (30s @ 16kHz)
  Stats: mean=0.002, std=0.15, range=[-0.98, 0.97]

Step 2: Mel Spectrogram
  Shape: [1, 80, 3000]
  Stats: mean=-4.2, std=2.1

Step 3: encoder.conv1
  Shape: [1, 384, 3000]
  Time: 2.3ms
  ⚠ Activation spike at position 1247 (value: 12.4)
```

For LLM inference, the equivalent is:
```
Step 1: TOKENIZE
  Input: "What is 2+2?"
  Output: [1, 3923, 374, 220, 17, 10, 17, 30]  (8 tokens)

Step 2: EMBED
  Shape: [8, 896]
  Stats: mean=0.002, std=0.45

Step 3: TRANSFORMER_BLOCK[0]
  Shape: [8, 896]
  Stats: mean=0.001, std=0.52
  Time: 1.2ms

... (layers 1-23) ...

Step 4: LM_HEAD
  Shape: [151936]
  Top-5: [(220, 12.4), (19, 11.2), (17, 10.8), ...]

Step 5: SAMPLE
  Selected: 220 (prob=0.42)

Step 6: DECODE
  Token 220 → "4"
```

### Phase 1: Add `generate_with_trace` to OwnedQuantizedModel

```rust
/// Generate tokens with full payload tracing (APR-TRACE-001)
pub fn generate_with_trace(
    &self,
    prompt: &[u32],
    config: &QuantizedGenerateConfig,
    tracer: &mut InferenceTracer,
) -> Result<Vec<u32>> {
    // Trace ENCODE (already done by caller, but record it)
    tracer.start_step(TraceStep::Tokenize);
    tracer.trace_encode("", prompt, self.config.vocab_size);

    let max_seq_len = prompt.len() + config.max_tokens;
    let mut cache = OwnedQuantizedKVCache::from_config(&self.config, max_seq_len);
    let mut tokens = prompt.to_vec();

    // Prefill with tracing
    for (pos, &token_id) in prompt.iter().enumerate() {
        tracer.start_step(TraceStep::Embed);
        let hidden = self.embed(token_id);
        tracer.trace_embed(1, self.config.hidden_dim, Some(&hidden));

        for (layer_idx, layer) in self.layers.iter().enumerate() {
            tracer.start_step(TraceStep::TransformerBlock);
            let output = self.forward_layer(layer, &hidden, &mut cache, pos)?;
            tracer.trace_layer(layer_idx, 0, Some(&output), 1, self.config.hidden_dim);
        }

        tracer.start_step(TraceStep::LmHead);
        let logits = self.lm_head(&hidden);
        tracer.trace_lm_head(0, &logits, self.config.vocab_size);
    }

    // Generate with tracing
    for gen_idx in 0..config.max_tokens {
        tracer.start_step(TraceStep::Sample);
        let next_token = Self::sample(&logits, config);
        tracer.trace_sample(gen_idx, &logits, next_token, config.temperature, config.top_k);

        tracer.start_step(TraceStep::Decode);
        tracer.trace_decode(gen_idx, next_token, "<needs_tokenizer>", self.config.vocab_size);

        if config.stop_tokens.contains(&next_token) {
            break;
        }
        tokens.push(next_token);

        // Forward next token with tracing (same pattern)
        // ...
    }

    Ok(tokens)
}
```

### Phase 2: Add `run_inference_traced` to infer/mod.rs

```rust
/// Run inference with full payload tracing (APR-TRACE-001)
pub fn run_inference_traced(config: &InferenceConfig) -> Result<(InferenceResult, InferenceTracer)> {
    let mut tracer = InferenceTracer::new(TraceConfig::enabled());
    tracer.set_model_info(ModelInfo {
        name: config.model_path.display().to_string(),
        // ... model metadata
    });

    // Use generate_with_trace instead of generate_with_cache
    let result = run_inference_with_tracer(config, &mut tracer)?;

    Ok((result, tracer))
}
```

### Phase 3: Connect `apr trace --payload` to traced inference

```rust
// In handle_special_modes():
if payload {
    // ACTUALLY RUN TRACED INFERENCE instead of just printing
    let config = InferenceConfig::new(path)
        .with_prompt("What is 2+2?")  // Test prompt
        .with_max_tokens(16);

    let (result, tracer) = run_inference_traced(&config)?;

    // Output the trace
    println!("{}", tracer.format_text());

    return Some(Ok(()));
}
```

## Acceptance Criteria

1. `apr trace model.gguf --payload` runs actual inference with a test prompt
2. Output shows INPUT → LAYER → OUTPUT for every step
3. Tensor stats (mean, std, min, max, NaN count) at each step
4. When garbage output occurs, the trace shows EXACTLY where it went wrong
5. GPU and CPU paths both emit trace events

## Files to Modify

1. `realizar/src/gguf_monolith.rs` - Add `generate_with_trace`
2. `realizar/src/infer/mod.rs` - Add `run_inference_traced`
3. `aprender/crates/apr-cli/src/commands/trace.rs` - Connect payload mode to traced inference
4. Tests for traced generation

## Priority

**P0 - STOP THE LINE**

Cannot debug GGUF garbage output (BUG-GGUF-001) without this capability. Flying blind.

## References

- APR-SPEC.md §4.17 (Trace Command)
- APR-TRACE-001 specification
- Toyota Way: Jidoka (stop the line when defects are detected)
- Popperian Falsification: Cannot falsify hypotheses without observability

## Labels

bug, P0, observability, tracing
